src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -83,181 +64,105 @@ describe('DOMPropertyOperations', () => {
+      var container = document.createElement('div');
+      ReactDOM.render(
+        <input type="radio" value="" onChange={function() {}} />,
+        container,
+      );
+      spyOn(container.firstChild, 'setAttribute');
+      ReactDOM.render(
+        <input type="radio" value={0} onChange={function() {}} />,
+        container,
+      );
+      expect(container.firstChild.setAttribute.calls.count()).toBe(1);
     });
 
     it('should always assign the value attribute for non-inputs', function() {
-      var stubNode = document.createElement('progress');
-      var stubInstance = {_debugID: 1};
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-
-      spyOn(stubNode, 'setAttribute');
-
-      DOMPropertyOperations.setValueForProperty(stubNode, 'value', 30);
-      DOMPropertyOperations.setValueForProperty(stubNode, 'value', '30');
-
-      expect(stubNode.setAttribute.calls.count()).toBe(2);
+      var container = document.createElement('div');
+      ReactDOM.render(<progress />, container);
+      spyOn(container.firstChild, 'setAttribute');
+      ReactDOM.render(<progress value={30} />, container);
+      ReactDOM.render(<progress value="30" />, container);
+      expect(container.firstChild.setAttribute.calls.count()).toBe(2);
     });
   });
 
   describe('deleteValueForProperty', () => {
-    var stubNode;
-    var stubInstance;
-
-    beforeEach(() => {
-      stubNode = document.createElement('div');
-      stubInstance = {_debugID: 1};
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-    });
-
     it('should remove attributes for normal properties', () => {
-      DOMPropertyOperations.setValueForProperty(stubNode, 'title', 'foo');
-      expect(stubNode.getAttribute('title')).toBe('foo');
-      expect(stubNode.title).toBe('foo');
-
-      DOMPropertyOperations.deleteValueForProperty(stubNode, 'title');
-      expect(stubNode.getAttribute('title')).toBe(null);
+      var container = document.createElement('div');
+      ReactDOM.render(<div title="foo" />, container);
+      expect(container.firstChild.getAttribute('title')).toBe('foo');
+      expect(container.firstChild.title).toBe('foo');
+      ReactDOM.render(<div />, container);
+      expect(container.firstChild.getAttribute('title')).toBe(null);
       // JSDOM does not behave correctly for attributes/properties
-      //expect(stubNode.title).toBe('');
+      //expect(container.firstChild.title).toBe('');
     });
 
     it('should not remove attributes for special properties', () => {
-      stubNode = document.createElement('input');
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-
-      stubNode.setAttribute('value', 'foo');
-
-      DOMPropertyOperations.deleteValueForProperty(stubNode, 'value');
-      // JSDOM does not behave correctly for attributes/properties
-      //expect(stubNode.getAttribute('value')).toBe('foo');
-      expect(stubNode.value).toBe('');
```

Comment:
Why did the test pass then, if it’s wrong? Is it because we were testing internals rather than public API?